packages/integrations/src/tracing.ts

```diff
@@ -0,0 +1,170 @@
+import { EventProcessor, Hub, Integration } from '@sentry/types';
+import { fill, getGlobalObject, isMatchingPattern, SentryError, supportsNativeFetch } from '@sentry/utils';
+
+/** JSDoc */
+interface TracingOptions {
+  tracingOrigins: Array<string | RegExp>;
+  traceXHR?: boolean;
+  traceFetch?: boolean;
+  autoStartOnDomReady?: boolean;
+}
+
+/**
+ * Tracing Integration
+ */
+export class Tracing implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public name: string = Tracing.id;
+
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Tracing';
+
+  /**
+   * If we have an xhr we need to store the url in the instance.
+   *
+   */
+  // @ts-ignore
+  private _xhrUrl?: string;
+
+  /**
+   * Constructor for Tracing
+   *
+   * @param _options TracingOptions
+   */
+  public constructor(private readonly _options: TracingOptions) {
+    if (!_options.tracingOrigins || !Array.isArray(_options.tracingOrigins) || _options.tracingOrigins.length === 0) {
+      throw new SentryError('You need to define `tracingOrigins` in the options. Set an array of urls to trace.');
```

Comment:
`of urls` or "patterns" maybe? It'd be nice to know what to place there from the error message.

Reply:
Also, are we sure we want to throw instead of log debug here? We could then just check for `Array.isArray` and provide sensible default options